src/Types/ContextFactory.php

```diff
@@ -70,6 +70,26 @@ public function createForNamespace($namespace, $fileContents)
                 case T_NAMESPACE:
                     $currentNamespace = $this->parseNamespace($tokens);
                     break;
+                case T_CLASS:
+                    // Fast-forward the iterator through the class so that any
+                    // T_USE tokens found within are skipped - these are not
+                    // valid namespace use statements so should be ignored.
+                    $braceLevel = 0;
+                    $firstBraceFound = false;
+                    while ($tokens->valid() && ($braceLevel > 0 || !$firstBraceFound)) {
+                        if ($tokens->current() === '{') {
```

Comment:
There is no class constant but it would not hurt to introduce it; though I do not consider it blocking for a merge